.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "*"
+
+jobs:
+  build-package:
+    name: Build & inspect our package.
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: hynek/build-and-inspect-python-package@v2
+
+  release-pypi:
+    name: Publish released package to pypi.org
+    permissions:
+      id-token: write
+    environment: release
```

Comment:
FTR, I recommend calling the environment `pypi` since it represents a deployment target which is PyPI. Also, it's nice to also show a URL on the UI.

Reply:
https://github.com/sarugaku/resolvelib/pull/177